ledfx/effects/metro.py

```diff
@@ -0,0 +1,104 @@
+import logging
+import timeit
+
+import numpy as np
+import voluptuous as vol
+
+from ledfx.color import parse_color, validate_color
+from ledfx.effects.temporal import TemporalEffect
+
+_LOGGER = logging.getLogger(__name__)
+
+# This effect is intended to assist in obvservations of synchronisity from the
+# browser to led strips and between led strips as it uses common timing between
+# all instances
+# control for time between flashes
+# control for mark space
+# control for count of divisions
+
+
+class MetroEffect(TemporalEffect):
+    NAME = "Metro"
+    CATEGORY = "Non-Reactive"
+    start_time = timeit.default_timer()
+
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Optional(
+                "speed",
+                default=20.0,
+                description="Speed of the effect",
+            ): vol.All(vol.Coerce(float), vol.Range(min=20, max=20)),
+            vol.Optional(
+                "pulse_period",
+                description="Time between flash in seconds",
+                default=1,
+            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=10)),
+            vol.Optional(
+                "pulse_ratio",
+                description="Flash to blank ratio",
+                default=0.3,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.1, max=1)),
+            vol.Optional(
+                "steps",
+                description="Steps of pattern division to loop",
+                default=4,
+            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=10)),
+            vol.Optional(
+                "background_color",
+                description="Background color",
+                default="#000000",
+            ): validate_color,
+            vol.Optional(
+                "flash_color",
+                description="Flash color",
+                default="#FFFFFF",
+            ): validate_color,
+        }
+    )
+
+    def __init__(self, ledfx, config):
+        super().__init__(ledfx, config)
+        self.step_count = 0
```

Comment:
Derive step_count directly from time passed / pulse period  % steps. So we can enforce synchronization between separate virtuals with same configs. Save time from forcing users to use copy virtual